### ISSUE TITLE
Sequence-classification recipe scales gradients by grad_accumulation_steps

### ISSUE THREAD

**Issue by kabirvashisht4-glitch** [NVIDIA-NeMo/Automodel#3829]
**Describe the bug**

`TrainFinetuneRecipeForSequenceClassification` accumulates gradients over
micro-batches without dividing by the micro-batch count, so the gradient
magnitude scales linearly with `grad_accumulation_steps`. Training with
gradient accumulation applies an effectively multiplied learning rate.

`nemo_automodel/recipes/llm/train_seq_cls.py` L90 and L227-242:

```python
self.loss_fn = torch.nn.CrossEntropyLoss()          # reduction="mean"
...
for batch in batches:                                # the grad-accumulation loop
    ...
    loss = self.loss_fn(logits, labels.view(-1))     # mean over THIS micro-batch
    ...
    (loss * self._get_dp_group_size(include_cp=True)).backward()
```

Each `.backward()` adds to the existing `.grad`, so accumulating `N`
micro-batch *means* gives `N x` the mean over the full accumulated batch.
Nothing divides by `len(batches)`.

The `* dp_group_size` factor is unrelated — it converts the DP-averaged
gradient back to a sum and is correct; the missing term is the accumulation
divisor.

**Contrast with the sibling recipes.** `recipes/llm/train_ft.py` normalizes by
`num_label_tokens`, the DP-reduced count over the whole accumulated batch
(L1211-1213, threaded into the loss at L1161), so its gradients are invariant to
how a global batch is split into micro-batches. `train_seq_cls.py` is the only
training recipe whose gradient depends on that split.

**Steps/Code to reproduce bug**

CPU only, no GPU or checkpoint needed. This reproduces the exact accumulation
the recipe performs:

```python
import torch, torch.nn as nn

torch.manual_seed(0)
V, C, N, B = 8, 3, 4, 2          # 4 micro-batches of 2 samples
lin = nn.Linear(V, C, bias=False)
loss_fn = nn.CrossEntropyLoss()
X = torch.randn(N * B, V)
Y = torch.randint(0, C, (N * B,))

# What the recipe does: backward each micro-batch mean, never divide by N.
lin.weight.grad = None
for i in range(N):
    loss_fn(lin(X[i * B:(i + 1) * B]), Y[i * B:(i + 1) * B]).backward()
g_recipe = lin.weight.grad.clone()

# Reference: one mean over the whole accumulated batch.
lin.weight.grad = None
loss_fn(lin(X), Y).backward()
g_ref = lin.weight.grad.clone()

print(g_recipe.norm() / g_ref.norm())
```

```
tensor(4.0)
```

The ratio is exactly the number of micro-batches.

**Consequence.** With `global_batch_size: 32` and `local_batch_size: 8`,
`grad_accumulation_steps` is 4 and every gradient is 4x too large — the same as
quadrupling the learning rate, silently. Raising `global_batch_size` to reduce
gradient noise instead *increases* the effective step size, which is the
opposite of the intent. Nothing errors; the run reports a normal loss, because
only the gradient is affected, not the logged loss value.

The bug is invisible at `grad_accumulation_steps == 1` (one micro-batch, ratio
1.0), so a smoke test with `global_batch_size == local_batch_size * dp_size`
passes.

**Expected behavior**

The accumulated gradient equals the gradient of the mean loss over the full
global batch, independent of how that batch is split into micro-batches —
matching `train_ft.py`.

**Environment overview**

- `main` at 3ddef9b1, CPU only.

**Additional context**

Dividing the per-micro-batch loss by `len(batches)` cancels the factor exactly
when micro-batches are equally sized, which is the normal case here (the
sampler drops the tail). Weighting each micro-batch by its own sample count
would also be exact for a ragged final batch — happy to do it either way; say
which you prefer.

While confirming this I also noticed `train_seq_cls.py` is the only recipe that
never calls `_set_moe_aux_loss_backward_scale` (`train_ft.py`, `llm/kd.py`,
`vlm/finetune.py` and `vlm/kd.py` all do). That one is **latent, not a live
bug**: the only sequence-classification architecture in `MODEL_ARCH_MAPPING` is
`LlamaBidirectionalForSequenceClassification`, which is dense, so no MoE
auxiliary loss currently flows through this recipe. Flagging it only because the
recipe does wire up `moe_mesh` and `moe_parallel_config`, so it would bite the
day a MoE sequence-classification model is registered.
